tests/lib/upload_manager_test.py

```diff
@@ -40,6 +40,16 @@ def test_single_local_path(self):
         self.assertTrue(os.path.exists(source))
         self.check_file_contains_string(self.bundle_location, 'testing')
 
+    def test_ignored_files(self):
+        dsstore_file = os.path.join(self.temp_dir, '.DS_Store')
+        self.write_string_to_file('testing', dsstore_file)
+        source = os.path.join(self.temp_dir, 'filename')
+        self.write_string_to_file('testing', source)
+        self.do_upload([self.temp_dir])
+        self.assertTrue(os.path.exists(os.path.join(self.bundle_location, 'filename')))
+        self.assertFalse(os.path.exists(os.path.join(self.bundle_location, '.DS_Store')))
```

Comment:
Add tests for the other patterns too

Reply:
Adding one for __MACOSX too, but I have no idea what the third pattern is for, and I've never had files besides __MACOSX and .DS_Store before, so it's not clear to me what to test that pattern with.